UnityProject/Assets/Scripts/Shuttles/CargoShuttle.cs

```diff
@@ -138,18 +141,21 @@ private Vector3 GetRandomFreePos()
 	{
 		int width = 2;
 		int height = 4;
+		int i = 0;
 
 		Vector3Int spawnPos;
-		while (true)
+		//temporary max crates in one
+		while (i < 40)
```

Comment:
why is there a hardcoded attempt limit? I mean, it can randomly choose to check same occupied position 40 times in a row and then give up

Reply:
I just couldn't come up with nice way of getting random position without repeats.
I mean, i can make double array check, but it wouldn't spawn crates randomly scattered around.

Reply:
Before starting to spawn orders you could calculate a List of non-occupied positions (`shuttleMatrix.IsPassableAt(LocalPos)`/`MatrixManager.IsPassableAt(WorldPos)`) within shuttle bounds, shuffle it, put it into a Queue and then Dequeue valid positions from that queue